tests/MongoDB.Bson.Tests/ObjectModel/ObjectIdTests.cs

```diff
@@ -14,25 +14,78 @@
 */
 
 using System;
+using System.Globalization;
 using System.Linq;
-using MongoDB.Bson;
+using FluentAssertions;
+using MongoDB.Bson.TestHelpers;
 using Xunit;
 
 namespace MongoDB.Bson.Tests
 {
     public class ObjectIdTests
     {
+        [Theory]
+        [InlineData(0xFFFFFE, 0xFFFFFF)]
+        [InlineData(0xFFFFFF, 0)]
+        [InlineData(0x1000000, 1)]
+        [InlineData(0x1FFFFFE, 0xFFFFFF)]
+        [InlineData(0x1FFFFFF, 0)]
+        [InlineData(0x2000000, 1)]
+        public void Ensure_that_increment_wraps_around_after_max_value(int seedIncrement, int expectedIncrement)
+        {
+            ObjectIdReflector.__staticIncrement(seedIncrement);
+            var objectId = ObjectId.GenerateNewId();
+            objectId.Increment.Should().Be(expectedIncrement);
+        }
+
+#if NET452
+        [Fact]
+        public void Ensure_that_machine_value_is_not_the_same_if_created_from_different_appdomain()
+        {
+            var objectIdCreatedInThisAppDomain = ObjectId.GenerateNewId();
+
+            var differentAppDomain = AppDomain.CreateDomain("DifferentObjectIdCreatorAppDomain", null, new AppDomainSetup { ApplicationBase = AppDomain.CurrentDomain.SetupInformation.ApplicationBase });
+            try
+            {
+                var objectIdCreatorType = typeof(ObjectIdCreator);
+                var objectIdCreator = (ObjectIdCreator)differentAppDomain.CreateInstanceAndUnwrap(objectIdCreatorType.Assembly.FullName, objectIdCreatorType.FullName);
+                var objectIdCreatedInDifferentAppDomain = objectIdCreator.CreateObjectId(); // CreateObjectId method runs in differentAppDomain
+#pragma warning disable 618
+                objectIdCreatedInDifferentAppDomain.Machine.Should().NotBe(objectIdCreatedInThisAppDomain.Machine);
```

Comment:
I'm no longer sure we want the AppDomainId to be part of the `Random`  value.

The spec says the `Random` value should be per **process**.

That might mean there is no way to test this.

Reply:
I tried to update this test to get `_random` value itself from ObjectId and it works in debug, but when I call it without debugging the random values for both cases are the same, not sure why. So, I removed this test if no objections